### PR TITLE
Added q! - command

### PR DESCRIPTION
**Pull request by Zarathustra2**
Hey,

I have added the _q!_ command, which discards all changes and quits the editor.
Furthermore, if someone now tries to quit with the _q_ command while changes have not been saved, it will fail and display an error message.

Since this is my first time contributing to a go open source project, I would appreciate any feedback!

Best regards,
Dario! :)

### CODE REVIEW

**Comment by itchyny** [comment]
This sounds nice. I'll take a look later.

File: cmdline/command.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ var commands = []command{
 
 	{"exi[t]", event.Quit},
 	{"q[uit]", event.Quit},
+	{"q!", event.ForceQuit},
```
Comment (itchyny) [inline cmdline/command.go]: I think it's better to support `q[uit]!` but I remember I didn't implement the parser to accept this so I'll fix this after merging.

Comment (Zarathustra2) [reply cmdline/command.go]: Yes, I tried _q[uit]!_ as well, but failed due to the parsing. Do you want me to add a failing test case for _q[uit]!_ ?

Comment (itchyny) [reply cmdline/command.go]: No, thanks. I'll take care of it.

**Comment by itchyny** [comment]
Your pull request updates unsavedChanges for only some limited operations. You can find all the changes by searching /b\.buffer\./, for example, `window#insert`, `window#replace`, `window#cut` or `window#paste`. Also, please rename `unsavedChanges` to `modified`, which reminds me `&modified` of Vim.

changes sample:
```diff
@@ -356,11 +356,13 @@ func (w *window) state(width, height int) (*state.WindowState, error) {
 func (w *window) insert(offset int64, c byte) {
 	w.buffer.Insert(offset, c)
 	w.changedTick++
+	w.modified = true
 }
 
 func (w *window) replace(offset int64, c byte) {
 	w.buffer.Replace(offset, c)
 	w.changedTick++
+	w.modified = true
 }
 
 func (w *window) undoReplace(offset int64) {
```

**Comment by Zarathustra2** [comment]
Hey @itchyny,

Thanks for reviewing my PR! :)

I have updated the files as requested, would you mind taking another look? 

Best regards!

**Comment by itchyny** [comment]
Great, thank you.

**Comment by itchyny** [comment]
Now that I improved the cmdline parser, you can use both `:q!`, `:quit!`.

**Comment by Zarathustra2** [comment]
Nice thanks! :) @itchyny

**Pull request merged**